Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #139220
* #139208
* __->__ #139081
* #139558
* #139090
* #139084

Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

swolchok committed:
[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel

Following the previous move of fp16_gemv_trans.

Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

malfet reviewed (changes requested):
This move will cause failures with compiled with SVE support

swolchok commented:
need to reorder this after https://github.com/pytorch/pytorch/pull/139090 and pull the fmadd(Vectorized<BFloat16>, Vectorized<BFloat16>, Vectorized<float>) from https://github.com/pytorch/pytorch/pull/139159 into this rev so that the vectorized tail loop will use SVE if appropriate rather than trying to mix Vectorized with NEON intrinsics.

swolchok committed:
reorder PRs; still working on SVE build issues on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
combine two PRs; still working on SVE build issues on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "Move bf16_gemv_trans to ReducedPrecisionFloatGemvFastPathKernel"


Following the previous move of fp16_gemv_trans.

Testing: Checked for performance regression with llm_benchmarks' `python benchmarks/benchmark_torch_mm.py llm`, didn't find one
Differential Revision: [D64930872](https://our.internmc.facebook.com/intern/diff/D64930872/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok commented:
some tests still in progress, but none of them are mps/aarch64 so I'm optimistic and re-requesting review.
